pkg/tsdb/sqleng/sql_engine_test.go

```diff
@@ -9,317 +9,308 @@ import (
+			require.Equal(t, float64(tSeconds)*1e3, fixtures[0].(float64))
+			require.Equal(t, float64(tSeconds)*1e3, fixtures[1].(float64))
+			require.Nil(t, fixtures[2])
 		})
+	})
 
-		Convey("Given row values with float64 as time columns", func() {
-			tSeconds := float64(dt.UnixNano()) / float64(time.Second)
-			tMilliseconds := float64(dt.UnixNano()) / float64(time.Millisecond)
-			tNanoSeconds := float64(dt.UnixNano())
-			var nilPointer *float64
-
-			fixtures := make([]interface{}, 7)
-			fixtures[0] = tSeconds
-			fixtures[1] = &tSeconds
-			fixtures[2] = tMilliseconds
-			fixtures[3] = &tMilliseconds
-			fixtures[4] = tNanoSeconds
-			fixtures[5] = &tNanoSeconds
-			fixtures[6] = nilPointer
-
-			for i := range fixtures {
-				ConvertSqlTimeColumnToEpochMs(fixtures, i)
+	t.Run("Given row with value columns", func(t *testing.T) {
+		intValue := 1
+		int64Value := int64(1)
+		int32Value := int32(1)
+		int16Value := int16(1)
+		int8Value := int8(1)
+		float64Value := float64(1)
+		float32Value := float32(1)
+		uintValue := uint(1)
+		uint64Value := uint64(1)
+		uint32Value := uint32(1)
+		uint16Value := uint16(1)
+		uint8Value := uint8(1)
+
+		fixtures := make([]interface{}, 24)
+		fixtures[0] = intValue
+		fixtures[1] = &intValue
+		fixtures[2] = int64Value
+		fixtures[3] = &int64Value
+		fixtures[4] = int32Value
+		fixtures[5] = &int32Value
+		fixtures[6] = int16Value
+		fixtures[7] = &int16Value
+		fixtures[8] = int8Value
+		fixtures[9] = &int8Value
+		fixtures[10] = float64Value
+		fixtures[11] = &float64Value
+		fixtures[12] = float32Value
+		fixtures[13] = &float32Value
+		fixtures[14] = uintValue
+		fixtures[15] = &uintValue
+		fixtures[16] = uint64Value
+		fixtures[17] = &uint64Value
+		fixtures[18] = uint32Value
+		fixtures[19] = &uint32Value
+		fixtures[20] = uint16Value
+		fixtures[21] = &uint16Value
+		fixtures[22] = uint8Value
+		fixtures[23] = &uint8Value
+
+		var intNilPointer *int
+		var int64NilPointer *int64
+		var int32NilPointer *int32
+		var int16NilPointer *int16
+		var int8NilPointer *int8
+		var float64NilPointer *float64
+		var float32NilPointer *float32
+		var uintNilPointer *uint
+		var uint64NilPointer *uint64
+		var uint32NilPointer *uint32
+		var uint16NilPointer *uint16
+		var uint8NilPointer *uint8
+
+		nilPointerFixtures := make([]interface{}, 12)
+		nilPointerFixtures[0] = intNilPointer
+		nilPointerFixtures[1] = int64NilPointer
+		nilPointerFixtures[2] = int32NilPointer
+		nilPointerFixtures[3] = int16NilPointer
+		nilPointerFixtures[4] = int8NilPointer
+		nilPointerFixtures[5] = float64NilPointer
+		nilPointerFixtures[6] = float32NilPointer
+		nilPointerFixtures[7] = uintNilPointer
+		nilPointerFixtures[8] = uint64NilPointer
+		nilPointerFixtures[9] = uint32NilPointer
+		nilPointerFixtures[10] = uint16NilPointer
+		nilPointerFixtures[11] = uint8NilPointer
+
+		t.Run("When converting values to float should return expected value", func(t *testing.T) {
+			for _, f := range fixtures {
```

Comment:
There should be a sub-test for each fixture instead, each fixture being a separate test case:
```suggestion
		for _, f := range fixtures {
				t.Run("", func(t *testing.T) {
```